pkg/externalLinks/ExternalLinksService.go

```diff
@@ -0,0 +1,311 @@
+				Name:             link.ExternalLinks.Name,
+				Url:              link.ExternalLinks.Url,
+				Active:           link.ExternalLinks.Active,
+				MonitoringToolId: link.ExternalLinks.ExternalLinksMonitoringToolId,
+				UpdatedOn:        link.UpdatedOn,
+			}
+		}
+		response[link.ExternalLinksId].ClusterIds = append(response[link.ExternalLinksId].ClusterIds, link.ClusterId)
+	}
+
+	for k, v := range response {
+		if _, ok := filterByCluster[k]; ok {
+			externalLinkResponse = append(externalLinkResponse, v)
+		} else if clusterId == 0 {
+			externalLinkResponse = append(externalLinkResponse, v)
+		}
+	}
+
+	//now add all the links which are not mapped to any clusters
+	additionalExternalLinks, err := impl.externalLinksRepository.FindAllNonMapped(mappedExternalLinksIds)
+	if err != nil {
+		impl.logger.Errorw("error in fetch all links", "err", err)
+		return nil, err
+	}
+	for _, link := range additionalExternalLinks {
+		providerRes := &ExternalLinksRequest{
+			Id:               link.Id,
+			Name:             link.Name,
+			Url:              link.Url,
+			Active:           link.Active,
+			MonitoringToolId: link.ExternalLinksMonitoringToolId,
+			ClusterIds:       []int{},
+			UpdatedOn:        link.UpdatedOn,
+		}
+		externalLinkResponse = append(externalLinkResponse, providerRes)
+	}
+
+	if externalLinkResponse == nil {
+		externalLinkResponse = make([]*ExternalLinksRequest, 0)
+	}
+	return externalLinkResponse, err
+}
+func (impl ExternalLinksServiceImpl) Update(request *ExternalLinksRequest) (*ExternalLinksCreateUpdateResponse, error) {
+	impl.logger.Debugw("link update request", "req", request)
+	externalLinks, err0 := impl.externalLinksRepository.FindOne(request.Id)
+	if err0 != nil {
+		impl.logger.Errorw("No matching entry found for update.", "id", request.Id)
+		return nil, err0
+	}
+	externalLinks.Name = request.Name
+	externalLinks.Url = request.Url
+	externalLinks.Active = true
+	externalLinks.ExternalLinksMonitoringToolId = request.MonitoringToolId
+	externalLinks.UpdatedBy = int32(request.UserId)
+	externalLinks.UpdatedOn = time.Now()
+	err := impl.externalLinksRepository.Update(&externalLinks)
+	if err != nil {
+		impl.logger.Errorw("error in updating link", "data", externalLinks, "err", err)
+		return nil, err
+	}
+
+	allExternalLinksMapping, err := impl.externalLinksClustersRepository.FindAllByExternalLinkId(request.Id)
+	if err != nil {
+		impl.logger.Errorw("error in fetching link", "data", externalLinks, "err", err)
+		return nil, err
+	}
+	for _, model := range allExternalLinksMapping {
+		model.Active = false
+		model.UpdatedBy = int32(request.UserId)
+		model.UpdatedOn = time.Now()
+		err := impl.externalLinksClustersRepository.Update(model)
+		if err != nil {
+			impl.logger.Errorw("error in updating clusters to false", "data", model, "err", err)
+			return nil, err
+		}
+	}
+	for _, requestedClusterId := range request.ClusterIds {
+		externalLinkClusterId := 0
+		var externalLinkCluster *ExternalLinksClusters
```

Comment:
Instead of this complex logic, shouldn't we directly insert the data?

Reply:
there is a use cases while updating items, add/remove in a single request possible here. so logic is okay.